### PR TITLE
Scheduling

### PR DESCRIPTION
**Pull request by jaredsnyder**
- Add decorators to schedule mobile and desktop tile outerbounds runs
- Added code to parse env vars for test_mode and write
- check to make sure it's running on production branch
- update CI to update the OB workflow upon merging to main

### CODE REVIEW

File: .circleci/workflows.yml [hunk 1]
```diff
@@ -111,9 +111,9 @@ workflows:
          filters:
            branches:
              only:
-               - main
+               - scheduling # REVERT BEFORE MERGE
```
Comment (chelseatroy) [inline .circleci/workflows.yml:114]: So, revert this before merging of course :P

File: .circleci/workflows.yml [hunk 2]
```diff
@@ -111,9 +111,9 @@ workflows:
...
       - update_mobile_tiles_scheduled_job:
          filters:
            branches:
              only:
-               - main
+               - scheduling # REVERT BEFORE MERGE
```
Comment (chelseatroy) [inline .circleci/workflows.yml:119]: Ditto :P

File: moz_forecasting/ad_tiles_forecast/flow.py [hunk 3]
```diff
@@ -164,7 +164,12 @@ def start(self):
         self.test_mode = self.test_mode.lower() == "true"
         logging.info(f"test_mode set to: {self.test_mode}")
 
-        if self.set_forecast_month is None or self.set_forecast_month == "null":
+        logging.info(f"Forecast month input as: {self.set_forecast_month}")
+        if (
+            self.set_forecast_month is None
+            or self.set_forecast_month == "null"
+            or self.set_forecast_month == ""
```
Comment (chelseatroy) [inline moz_forecasting/ad_tiles_forecast/flow.py:171]: Ultra minor, but "" evaluates to False in Python, so you might be able to get away with just `if not self.set_forecast_month or self.set_forecast_month == "null"` here

**Review by chelseatroy — COMMENTED** [review]
See comments. Would approve except there are some comments in here that say "REVERT BEFORE MERGE" lol

File: moz_forecasting/ad_tiles_forecast/flow.py [hunk 4]
```diff
@@ -103,6 +107,7 @@ def get_direct_allocation_df(
     return direct_allocation_df
 
 
+@schedule(cron="0 1 3 * ? *", timezone="Etc/UTC")
```
Comment (chelseatroy) [inline moz_forecasting/ad_tiles_forecast/flow.py:110]: _Maybe_ worth commenting above this "1 AM UTC on the 3rd of every month"?

File: moz_forecasting/ad_tiles_forecast/flow.py [hunk 5]
```diff
@@ -114,13 +119,15 @@ class AdTilesForecastFlow(FlowSpec):
         default="moz_forecasting/ad_tiles_forecast/config_2025_planning_stretch.yaml",
     )
 
-    test_mode = Parameter(
+    test_mode_param = Parameter(
         name="test_mode",
         help="indicates whether or not run should affect production",
-        default=True,
+        default="true",
```
Comment (chelseatroy) [inline moz_forecasting/ad_tiles_forecast/flow.py:125]: This needs to be a string and not the Boolean? That's a shame

File: moz_forecasting/ad_tiles_forecast/flow.py [hunk 6]
```diff
@@ -135,15 +142,40 @@ def start(self):
 
         You can use it for collecting/preprocessing data or other setup tasks.
         """
+        # for scheduled flows set write with env var SCH_METAFLOW_PARAM_WRITE
+        write_envar = os.environ.get("SCH_METAFLOW_PARAM_WRITE")
+        if write_envar is not None:
+            self.write = write_envar
+        else:
+            self.write = self.write_param
+        # convert to boolean because parameters are passed as strings
```
Comment (chelseatroy) [inline moz_forecasting/ad_tiles_forecast/flow.py:151]: Answers my question above

File: moz_forecasting/ad_tiles_forecast/flow.py (hunk 5 above)
Comment (jaredsnyder) [reply moz_forecasting/ad_tiles_forecast/flow.py:125]: Yeah I needed consistency between the commandline and the env vars I had to use for setting variables in the prod flow.  Kind of annoying.

File: moz_forecasting/ad_tiles_forecast/flow.py (hunk 4 above)
Comment (jaredsnyder) [reply moz_forecasting/ad_tiles_forecast/flow.py:110]: Yeah that's a good idea

File: moz_forecasting/ad_tiles_forecast/flow.py (hunk 3 above)
Comment (jaredsnyder) [reply moz_forecasting/ad_tiles_forecast/flow.py:171]: Yeah that is better, gotta be pythonic

**Comment by jaredsnyder** [comment]
Confirmed prod runs work on outerbounds: https://ui.desertowl.obp.outerbounds.com/dashboard/flows/p/revenue

**Comment by chelseatroy** [comment]
Wahooo! Excellent engineering work @jaredsnyder. :shipit:

**Review by chelseatroy — APPROVED** [review]

**Pull request merged**